src/VisualStudio/Core/Impl/ProjectSystem/CPS/TempPECompiler.cs

```diff
@@ -49,16 +49,31 @@ public async Task<bool> CompileAsync(IWorkspaceProjectContext context, string ou
                 cancellationToken.ThrowIfCancellationRequested();
             }
 
-            var options = project.LanguageServices.GetRequiredService<ICompilationFactoryService>().GetDefaultCompilationOptions()
+            // We need to inherit most of the projects options, mainly for VB (RootNamespace, GlobalImports etc.), but we need to override about some specific things surrounding the output
+            var options = project.CompilationOptions
                     // copied from the old TempPE compiler used by legacy, for parity.
                     // See: https://github.com/dotnet/roslyn/blob/fab7134296816fc80019c60b0f5bef7400cf23ea/src/VisualStudio/CSharp/Impl/ProjectSystemShim/TempPECompilerService.cs#L58
                     .WithAssemblyIdentityComparer(DesktopAssemblyIdentityComparer.Default)
                     .WithSourceReferenceResolver(SourceFileResolver.Default)
                     .WithXmlReferenceResolver(XmlFileResolver.Default)
-                    // we always want to produce a DLL
-                    .WithOutputKind(OutputKind.DynamicallyLinkedLibrary);
+                    // We always want to produce a debug, AnyCPU DLL
+                    .WithOutputKind(OutputKind.DynamicallyLinkedLibrary)
+                    .WithPlatform(Platform.AnyCpu)
+                    .WithOptimizationLevel(OptimizationLevel.Debug)
+                    // Turn off any warnings as errors just in case
+                    .WithGeneralDiagnosticOption(ReportDiagnostic.Suppress)
+                    .WithReportSuppressedDiagnostics(false)
+                    .WithSpecificDiagnosticOptions(null)
+                    // Turn off any signing and strong naming
```

Comment:
Are you sure? This can break projects if your TempPE needs to consume a type that is coming from an internals visible to.

Reply:
This is matching legacy, and matching the previous behaviour here, so whilst what you're saying is a possible breaking point it wouldn't be a supported scenario by either project system.

Reply:
I don't know what your scenarios are; all I do know is I've seen plenty of people think signing/strong naming only applies to emitting binaries but it does have a fundamental impact on semantics. As long as your scenarios don't run into problems there, this is fine.